src/lib/dif.pl

```diff
@@ -23,6 +23,29 @@
     put_dif_att(Var, X, Y),
     dif_set_variables(Vars, X, Y).
 
+remove_goal([], _, []).
+remove_goal([G0|G0s], Goal0, Goals) :-
+    (   G0 == Goal0 ->
+        remove_goal(G0s, Goal0, Goals)
+    ;   Goals = [G0|Goals1],
+        remove_goal(G0s, Goal0, Goals1)
+    ).
+
+vars_remove_goal([], _).
+vars_remove_goal([Var|Vars], Goal0) :-
+    get_atts(Var, +dif(Goals0)),
```

Comment:
If you want, you can omit the `+` wrapper, and write this equivalently as `get_atts(Var, dif(Goals0))`.

Reply:
Yeah, I know. I don't know if there is actually any performance implication, but aesthetically I think I prefer this way because it's obviously not defaulty. Also, the rest of this file already used the `+` everywhere anyway, in `put_dif_att/3` for example.

Reply:
Very valid reasons for retaining the wrapper!